localstack/services/awslambda/provider.py

```diff
@@ -3155,19 +3172,20 @@ def add_layer_version_permission(
         organization_id: OrganizationId = None,
         revision_id: String = None,
     ) -> AddLayerVersionPermissionResponse:
-        # TODO: add layer ARN as layer_name support
-
-        layer_version_arn = api_utils.layer_version_arn(
-            layer_name, context.account_id, context.region, str(version_number)
-        )
+        region_name, account_id, layer_n, _ = LambdaProvider._resolve_layer(layer_name, context)
```

Comment:
Why use `layer_n` instead of `layer_name` here suddenly? I actually had to think a second there what that is supposed to mean.

Reply:
`layer_name` is the function argument and can contains either the layer name or layer ARN. This is user-provided and error messages are generated accordingly.

`layer_n` contains the layer name and is named so to avoid overwriting :top: